# INFERENTZIA ESTATISTIKOA: Laugarren praktika

## Hipotesi kontrasteak: Bi populazioen konparaketak

Bigarren praktikarekin (estimazioa eta KTak) harreman handiko praktika da hau. Izan ere, praktika horretan erabili genituen komandoak erabiliko ditugu eta ikusiko dugu nola aldatu behar diren hipotesi kontrasteak burutzeko.


In [10]:
# txt datuak irakurri

Datuak <-read.table(file="altuerak.txt", header=TRUE, dec=",", sep="\t")

# Hurrengo paketea instalatu eta kargatu behar duzue 
# install.packages("EnvStats")
# library(EnvStats)


## Bi populazio askeren bariantzen zatidura

### Gizonezkoen eta emakumezkoen altueren bariantzen zatiduraren HK

Emakumezkoen altueren bariantza gizonezkoena izatearen hipotesia kontrastatuko dugu:

$H_O:\ \sigma_1^2 \geq \sigma_2^2$

$H_1:\ \sigma_1^2 < \sigma_2^2$

In [11]:
emakumezkoa <-Datuak$ALTUERA[Datuak$SEXUA=="emakumezkoa"]#c(103,94,110,87,98)       
gizonezkoa  <-Datuak$ALTUERA[Datuak$SEXUA=="gizonezkoa"]#c(97, 82,123,92,175,88,118)
var.test(emakumezkoa,gizonezkoa,ratio=1, alternative="less")


	F test to compare two variances

data:  emakumezkoa and gizonezkoa
F = 0.88431, num df = 81, denom df = 88, p-value = 0.288
alternative hypothesis: true ratio of variances is less than 1
95 percent confidence interval:
 0.000000 1.269746
sample estimates:
ratio of variances 
         0.8843092 


#### Azalpena:
Planteatutako HKaren estatistikoa 0.88431 da, askatasun graduak 81 eta 88 eta p-balioa 0.288. Hortaz, ezin dugu ondorioztatu emakumezkoen barintza gizonezkoena baino handiagoa denik $\%5$eko adierazgarritasun mailaz.

#### Oharrak: 
* $H_O:\ \sigma_1^2 \geq 2 \sigma_2^2$,
$H_1:\ \sigma_1^2 < 2 \sigma_2^2$ kontrastatzeko, ratio=1 jarri ordez ratio=2 jarri behar da.
* Aldebiko eskuinetiko kontrastea egiteko, alternative="greater" jarri behar da
* Aldebateko kontrastea egiteko, alternative="two.sides" jarri behar da 

In [12]:
# ondoko HKa aurrekoaren berdina da. Ikusten duzu arrazoia?
var.test(gizonezkoa,emakumezkoa,ratio=1, alternative="greater")


	F test to compare two variances

data:  gizonezkoa and emakumezkoa
F = 1.1308, num df = 88, denom df = 81, p-value = 0.288
alternative hypothesis: true ratio of variances is greater than 1
95 percent confidence interval:
 0.7875588       Inf
sample estimates:
ratio of variances 
          1.130826 


## Bi populazio askeren batezbestekoen konparaketa

### Gizonezkoen eta emakumezkoen altueren batezbestekoaren diferentziaren

Emakumezkoen altueren batezbestekoa gizonezkoena baino handiagoa izatearen hipotesia kontrastatuko dugu:

$H_O:\ \mu_1 \leq \mu_2$

$H_1:\ \mu_1 > \mu_2^2$

In [13]:
t.test(emakumezkoa,gizonezkoa,mu=0,alternative="greater",var.equal=TRUE)


	Two Sample t-test

data:  emakumezkoa and gizonezkoa
t = -4.0665, df = 169, p-value = 1
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -2.790305       Inf
sample estimates:
mean of x mean of y 
 165.7805  167.7640 


#### Azalpena:
Emakumezkoen batezbesteko altueraren estimazioa: 165.7805

Gizonezkoen batezbesteko altueraren estimazioa: 165.7805

Kenketaren HKa: estatistikoaren balioa -4.0665 da, askatasun graduak 169 eta p-balioa 1 da. Hortaz, ezin dugu errefusatu emakumeen batezbestekoa gizonezkoena baino txikiagoa edo berdin dela.


#### Oharrak: 
* $H_O:\ \mu_1 - \mu_2 \leq 40$,
$H_1:\ \mu_1 - \mu_2 > 40 $ kontrastatzeko, mu=0 jarri ordez mu=40 jarri behar da.
* Aldebiko ezkerretiko kontrastea egiteko, alternative="less" jarri behar da
* Aldebateko kontrastea egiteko, alternative="two.sides" jarri behar da 
* Gogoratu var.equal=TRUE jarri dugula aurreko adibidean lortu duguna kontuan hartu dugulako.

In [14]:
# Ondoko HKa berdina da. Ikusten duzu zergatik?
t.test(gizonezkoa,emakumezkoa,mu=0,alternative="less",var.equal=TRUE)


	Two Sample t-test

data:  gizonezkoa and emakumezkoa
t = 4.0665, df = 169, p-value = 1
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
     -Inf 2.790305
sample estimates:
mean of x mean of y 
 167.7640  165.7805 


## Binakako datuen KTa

### Aitaren eta amaren altueren arteko diferentzia

Izan bedi D=Aitarena-Amarena (alegia, aitaren altuera ken amaren altuera). D aldagaiaren batezbestekoa positiboa izatearen hipotesia (alegia, aitaren altuera amarena baino handiagoa izatearen hipotesia ) kontrastatuko dugu:

$H_0:\ \mu_D \leq 0 $

$H_1:\ \mu_D > 0$

In [15]:
t.test(Datuak$AITARENA,Datuak$AMARENA, mu=0, alternative="greater")


	Welch Two Sample t-test

data:  Datuak$AITARENA and Datuak$AMARENA
t = 23.914, df = 316.34, p-value < 2.2e-16
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 10.52428      Inf
sample estimates:
mean of x mean of y 
 172.1871  160.8830 


Binakako datuen HKa: estatistikoaren balioa 23.914 da, askatasun graduak 316.34 eta p-balioa 0.0001 baino txikiagoa da. Hortaz, aitaren altuera amarena baino handiagoa dela ondorioztatzen dugu.


#### Oharrak: 
* $H_O:\ \mu_D  \leq 10$,
$H_1:\ \mu_D > 10 $ kontrastatzeko, mu=0 jarri ordez mu=10 jarri behar da.
* Aldebiko ezkerretiko kontrastea egiteko, alternative="less" jarri behar da
* Aldebateko kontrastea egiteko, alternative="two.sides" jarri behar da 

## Bi populazio binomialen proportzioen konparaketa

### Emakumeen eta gizonen bizkaitarren proportzioen diferentziaren HKa

Emakume bizkaitarren proportzioa gizonezkoena baino handiagoa izatearen hipotesia kontrastatuko dugu:

$H_0:\ p_1 \leq p_2 $

$H_1:\ p_1 > p_2$

non $p_1$  den eta $p_2$ emakumezkoen artean bizkaitarren proportzioa

In [27]:
# zenbat bizkaitar diren emakume eta gizon zenbatzen dugu
Datuak.bizkaia <- subset(Datuak,JAIOTERRIA==1)
emakume.bizkaia <- subset(Datuak.bizkaia,SEXUA=="emakumezkoa")
gizona.bizkaia <- subset(Datuak.bizkaia,SEXUA=="gizonezkoa")
n_emaku_biz <- length(emakume.bizkaia$ALTUERA)
n_gizon_biz <- length(gizona.bizkaia$ALTUERA)
# zenbat emakume eta gizon zenbatzen dugu
Datuak.emaku <- subset(Datuak,SEXUA=="emakumezkoa")
n_emaku <- length(Datuak.emaku$ALTUERA)
Datuak.gizon <- subset(Datuak,SEXUA=="gizonezkoa")
n_gizon <- length(Datuak.gizon$ALTUERA)
# testa egiten dugu
prop.test(c(n_emaku_biz,n_gizon_biz),c(n_emaku,n_gizon),alternative="greater",conf.level=0.95)
#guztiak
#binom.test(kopuru.bizkaia,guztiak,conf.level=0.95)


	2-sample test for equality of proportions with continuity correction

data:  c(n_emaku_biz, n_gizon_biz) out of c(n_emaku, n_gizon)
X-squared = 0.43037, df = 1, p-value = 0.2559
alternative hypothesis: greater
95 percent confidence interval:
 -0.06365059  1.00000000
sample estimates:
   prop 1    prop 2 
0.2560976 0.2022472 


Bi populazio binomial askeen arrakasta proportzioen alderaketaren ondorioa: estatistikoaren balioa 0.43037 da eta p-balioa 0.2559 da. Hortaz, ezin dugu ondorioztatu gizonezkoen artean bizkaitarren proportzioa emakumezkoen artekoa baino handiagoa denik.


#### Oharrak: 
* Aldebiko ezkerretiko kontrastea egiteko, alternative="less" jarri behar da
* Aldebateko kontrastea egiteko, alternative="two.sides" jarri behar da 